src/main/java/org/micks/javasamples/set_samples/WordRepository.java

```diff
@@ -0,0 +1,53 @@
+package org.micks.javasamples.set_samples;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class WordRepository {
+
+    private final Set<@NotNull String> words = new HashSet<>();
+
+    public void addWord(String word) {
+        if (word != null) {
+            words.add(word.toLowerCase());
+        }
+    }
+
+    public void addWords(Collection<String> newWords) {
+        for (String word : newWords) {
+            if (word != null) {
+                words.add(word.toLowerCase());
```

Comment:
I think there should be method like addAll() to? This way it will be simpler, no need for iterating manually with for loop.